File: GestionCours/src/Entity/Publicite.php
<?php

namespace App\Entity;

use App\Repository\PubliciteRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Vich\UploaderBundle\Mapping\Annotation as vich;

/**
 * @ORM\Entity(repositoryClass=PubliciteRepository::class)
 */
class Publicite
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\Url(
     *    relativeProtocol = true
     * )
     */
    private $lienPublicite;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $imagePublicite;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\Url(
     *    relativeProtocol = true
     * )
     */
    private $videoPublicite;

    /**
     * @ORM\Column(type="integer", nullable=true)
     */
    private $nbrAffichage;

    /**
     * @ORM\Column(type="integer", nullable=true)
     */
    private $nbrClick;

    /**
     * @ORM\ManyToOne(targetEntity=Cible::class, inversedBy="publicites")
     * @ORM\JoinColumn(nullable=false)({
     *   @ORM\JoinColumn(name="cible_id", referencedColumnName="id")
     */
    private $cible;


    protected $captchaCode;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getLienPublicite(): ?string
    {
        return $this->lienPublicite;
    }

    public function setLienPublicite(string $lienPublicite): self
    {
        $this->lienPublicite = $lienPublicite;

        return $this;
    }

    public function getImagePublicite(): ?string
    {
        return $this->imagePublicite;
    }

    public function setImagePublicite(string $imagePublicite): self
    {
        $this->imagePublicite = $imagePublicite;

        return $this;
    }

    public function getVideoPublicite(): ?string
    {
        return $this->videoPublicite;
    }

    public function setVideoPublicite(string $videoPublicite): self
    {
        $this->videoPublicite = $videoPublicite;

        return $this;
    }

    public function getNbrAffichage(): ?int
    {
        return $this->nbrAffichage;
    }

    public function setNbrAffichage(?int $nbrAffichage): self
    {
        $this->nbrAffichage = $nbrAffichage;

        return $this;
    }

    public function getNbrClick(): ?int
    {
        return $this->nbrClick;
    }

    public function setNbrClick(?int $nbrClick): self
    {
        $this->nbrClick = $nbrClick;

        return $this;
    }

    public function getCible(): ?cible
    {
        return $this->cible;
    }

    public function setCible(?cible $cible): self
    {
        $this->cible = $cible;

        return $this;
    }
    public function getCptchaCode()
    {
        return $this->captchacode;
    }
    public function CptchaCode($captchacode): self
    {
        $this->captchacode = $captchacode;

    }
}

File: 9ariniPi/src/entities/Publicite.java
 /*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package entities;

import java.util.Date;






/**
 *
 * @author pc-asus
 */
public class Publicite {
   private int id ;
   private String video_publicite ;
   
 
   private String lien_publicite  ;
 
   private int nbr_click  ;
   private int nbr_affichage;
   private String image_publicite ;
   private int cible_id;

    public int getCible_id() {
        return cible_id;
    }

    public void setCible_id(int cible_id) {
        this.cible_id = cible_id;
    }

    

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getVideo_publicite() {
        return video_publicite;
    }

    public void setVideo_publicite(String video_publicite) {
        this.video_publicite = video_publicite;
    }

    public Publicite(int id, String video_publicite, String lien_publicite, String image_publicite) {
        this.id = id;
        this.video_publicite = video_publicite;
        this.lien_publicite = lien_publicite;
        this.image_publicite = image_publicite;
    }

   

  

    

   

    public String getLien_publicite() {
        return lien_publicite;
    }

    public void setLien_publicite(String lien_publicite) {
        this.lien_publicite = lien_publicite;
    }

    public int getNbr_click() {
        return nbr_click;
    }

    public void setNbr_click(int nbr_click) {
        this.nbr_click = nbr_click;
    }

    public int getNbr_affichage() {
        return nbr_affichage;
    }

    public void setNbr_affichage(int nbr_affichage) {
        this.nbr_affichage = nbr_affichage;
    }

    public String getImage_publicite() {
        return image_publicite;
    }

    public void setImage_publicite(String imagePublicite) {
        this.image_publicite = imagePublicite;
    }

    public Publicite() {
    }

   

   
    ///////////////////////////////////

    public Publicite(String video_publicite, String lien_publicite, String image_publicite,int cible_id) {
        this.video_publicite = video_publicite;
        this.lien_publicite = lien_publicite;
        this.image_publicite = image_publicite;
        this.cible_id=cible_id;
    }
    public Publicite(String video_publicite, String lien_publicite, String image_publicite) {
       
        this.video_publicite = video_publicite;
        this.lien_publicite = lien_publicite;
        this.image_publicite = image_publicite;
        
    }
    public Publicite(String video_publicite) {
        this.video_publicite = video_publicite;
    }

    public Publicite(int id, String video_publicite,  String lien_publicite, int nbr_click, int nbr_affichage, String image_publicite) {
        this.id = id;
        this.video_publicite = video_publicite;
       
        this.lien_publicite = lien_publicite;
        this.nbr_click = nbr_click;
        this.nbr_affichage = nbr_affichage;
        this.image_publicite = image_publicite;
    }

    public Publicite(String video_publicite, String lien_publicite, int nbr_click, int nbr_affichage, String image_publicite) {
        this.video_publicite = video_publicite;
       
        this.lien_publicite = lien_publicite;
        this.nbr_click = nbr_click;
        this.nbr_affichage = nbr_affichage;
        this.image_publicite = image_publicite;
    }

    public Publicite(int id, String genre, int nbr_click, int nbr_affichage) {
        this.id = id;
       
        this.nbr_click = nbr_click;
        this.nbr_affichage = nbr_affichage;
    }

  
    

    @Override
    public String toString() {
        return "Publicite{" + "video_publicite=" + video_publicite + ", lien_publicite=" + lien_publicite + ", image_publicite=" + image_publicite + '}';
    }

   
    
    
}
